Sed sees a end of string in the whatever encoding, causing a compilation
failure, so just don't look for a pattern but replace the line with an
empty one (to respect the cpp lineno)

--- ocamldoc/remove_DEBUG
+++ ocamldoc/remove_DEBUG
@@ -18,4 +18,4 @@
 # respecting the cpp # line annotation conventions
 
 echo "# 1 \"$1\""
-LC_ALL=C sed -e '/DEBUG/s/.*//' "$1"
+LC_ALL=C sed -e '/DEBUG/c\ ' "$1"
